src/internal/pfsdb/branches.go

```diff
@@ -8,6 +8,22 @@ import (
 	"github.com/pachyderm/pachyderm/v2/src/pfs"
 )
 
+// SliceDiff takes two slices and returns the elements in the first slice that are not in the second slice.
+// TODO this can be moved to a more generic package.
+func SliceDiff[K comparable, V any](a []V, b []V, key func(V) K) []V {
```

Comment:
[nit] you may be able to declare this like: 
```suggestion
func SliceDiff[K comparable, V any](a, b []V, key func(V) K) []V {
```